### PR TITLE
[xbmc][fix] Use fmt for logging as well.

### PR DESCRIPTION
**Pull request by Paxxi**
Use fmt for logging as well. Changing the CLog methods to variadic templates.
Fixes a lot of pointer casts that fmt dislikes when using %p.
Removes CWin32Log rather than implementing LogW in the new way.
Hides CLogGlobals in the cpp file, getting rid of criticalsection in header.

Fixes https://trac.kodi.tv/ticket/17565
Likely doesn't build on non-windows yet

### CODE REVIEW

File: xbmc/windowing/rpi/RPIUtils.cpp [hunk 1]
```diff
@@ -560,8 +560,8 @@ void CRPIUtils::GetSupportedModes(HDMI_RES_GROUP_T group, std::vector<RESOLUTION
         m_desktopRes = res;
 
       AddUniqueResolution(res, resolutions);
-      CLog::Log(LOGDEBUG, "EGL mode %d: %s (%.2f) %s%s:%x\n", i, res.strMode.c_str(), res.fPixelRatio,
-          tv->native ? "N" : "", tv->scan_mode ? "I" : "", tv->code);
+      CLog::Log(LOGDEBUG, "EGL mode %d: %s (%.2f) %s%s:%x\n", i, res.strMode, res.fPixelRatio,
+          tv->native ? "N" : "", tv->scan_mode ? "I" : "", int(tv->code));
```
Comment (garbear) [inline xbmc/windowing/rpi/RPIUtils.cpp]: static_cast please

Also removing the .c_str() is compatible with `%s`? Or do we need `{0:s}`-style format?

Comment (Paxxi) [reply xbmc/windowing/rpi/RPIUtils.cpp]: it's compatible, fmt supports both formats.

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp [hunk 2]
```diff
@@ -225,10 +230,15 @@ void CMMALVideo::dec_output_port_cb(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *buf
...
-          m_decoded_width, m_decoded_height, m_decoded_aligned_width, m_decoded_aligned_height, (char*)&omvb->Encoding());
+      CLog::Log(LOGDEBUG, LOGVIDEO,
+                "%s::%s - omvb:%p mmal:%p len:%u dts:%.3f pts:%.3f flags:%x:%x pool:%p %dx%d "
+                "(%dx%d) %dx%d (%dx%d) enc:%.4s",
+                CLASSNAME, __func__, static_cast<void*>(buffer), static_cast<void*>(omvb),
+                buffer->length, buffer->dts * 1e-6, buffer->pts * 1e-6, buffer->flags,
+                buffer->type->video.flags, static_cast<void*>(m_pool.get()) omvb->Width(),
+                omvb->Height(), omvb->AlignedWidth(), omvb->AlignedHeight(), m_decoded_width,
+                m_decoded_height, m_decoded_aligned_width, m_decoded_aligned_height,
+                (char*)&omvb->Encoding());
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp]: Might as well update this one too

Comment (Paxxi) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp]: Not entirely sure what that cast does, from what I can tell Encoding returns a uint32_t so casting to (char*) is weird.

I've changed things sporadically as I've fixed log statements that uses `%p` so there's a lot of things I haven't touched that should be fixed later on.

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by Paxxi** [comment]
@rechi I think this is ready for review.

File: xbmc/windowing/rpi/RPIUtils.cpp (hunk 1 above)
Comment (Paxxi) [reply xbmc/windowing/rpi/RPIUtils.cpp]: Oh and the int() is not a cast, it creates a temporary because the type is a bitfield which fmt can't handle

**Comment by FernetMenta** [comment]
this change makes pvr crash, in particular: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp#L208

**Comment by Paxxi** [comment]
Thanks! fritsch mentioned a crash earlier today but didn't find anything by going through the changes. Fix coming up

**Pull request merged**